### ISSUE TITLE
Transformer outputs incorrect paths when in subdirectory

### ISSUE THREAD

**Issue by donny-dont** [dart-lang/polymer-dart#34]
If you are building the contents of web and an entry point is in a sub directory of web then the transformer will fail to output the right path when executing pub build.

Example transformer.

```yaml
transformers:
- polymer:
    entry_points:
    - web/index.html     # Builds fine
    - web/sub/index.html # Builds incorrectly
```

The following is the behavior of the transformer in the sub folder.

`pub build`
* packages/web_components/webcomponents.min.js :-1: 
* packages/web_components/dart_support.js :-1: 
* ../packages/polymer/src/js/polymer/polymer.min.js :+1: 

`pub build --mode=debug`
* packages/web_components/webcomponents.js :-1: 
* packages/web_components/dart_support.js :-1: 
* ../packages/polymer/src/js/polymer/polymer.min.js :+1: 
* packages/browser/dart.js :-1:

**Comment by jakemac53** [comment]
Have you done a pub upgrade? A fix went out for a very similar issue
recently.

**Comment by donny-dont** [comment]
Yep I'm all up to date.

```
Resolving dependencies...
  analyzer 0.22.4 (0.24.0 available)
  args 0.12.2+6 (0.13.0 available)
  barback 0.15.2+4
  browser 0.10.0+2
  cli_util 0.0.1+1
  code_transformers 0.2.7+1
  collection 1.1.0
  csslib 0.11.0+4
  dart_style 0.1.3 (0.1.7 available)
  glob 1.0.4
  html5lib 0.12.0
  initialize 0.5.1+5
  logging 0.9.3
  matcher 0.11.4+3 (0.12.0-alpha.0 available)
  mock 0.11.0+2
  observe 0.13.0
  path 1.3.3
  polymer 0.16.0+1
  polymer_expressions 0.13.1
  pool 1.0.1
  smoke 0.2.1+1 (0.3.1 available)
  source_maps 0.10.0+1
  source_span 1.0.3
  stack_trace 1.2.4
  string_scanner 0.1.3
  template_binding 0.14.0+1
  unittest 0.11.5+4 (0.12.0-alpha.1 available)
  utf 0.9.0+2
  watcher 0.9.5
  web_components 0.10.5+3
  when 0.2.0
  which 0.1.3
  yaml 2.1.2
No dependencies changed.
```

**Comment by jakemac53** [comment]
Actually, shouldn't there be a packages symlink in all folders under web?

**Comment by donny-dont** [comment]
Yes there is. The issue is that when you do a build packages is only in web/packages. There's no web/sub/packages. The bug is that the transformer doesn't realize this.

**Comment by donny-dont** [comment]
Not sure if it matters but this is on Windows 7 x64

**Comment by jakemac53** [comment]
I just tried it and interestingly in the built output I got a packages folder in web/foo, but it wasn't a symlink to ../packages, and it only had one package in it (the `browser` package).

**Comment by donny-dont** [comment]
That's weird. I have only seen the packages folder appear in build/web/. It makes sense that pub operates that way with how browsers cache files.

**Comment by jakemac53** [comment]
fix out here https://codereview.chromium.org/1011803002/

**Comment by jakemac53** [comment]
published fix as 0.16.0+2

**Comment by donny-dont** [comment]
:+1: just tested and works as advertised now.
